### PR TITLE
[quant] conv_transpose graph patterns

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45078 [quant] conv_transpose graph patterns**

Differential Revision: [D23821580](https://our.internmc.facebook.com/intern/diff/D23821580)

### CODE REVIEW

**Comment by z-a-f** [comment]
Resubmission of the #43199

File: test/quantization/test_quantize_jit.py [hunk 1]
```diff
@@ -3124,6 +3124,34 @@ def test_conv(self):
                 inplace=False)
             self.assertEqual(model_quantized(self.img_data_2d[0][0]), result_eager)
 
+    @override_qengines
+    def test_conv_transpose(self):
+        r"""Compare the result of quantizing conv_transpose layer in
+        eager mode and graph mode
+        """
+        if qengine_is_qnnpack():  # Currently only qnnpack is supported
```
Comment (vkuzo) [inline test/quantization/test_quantize_jit.py:3132]: optional nit to save an indent:
```
if not qengine_is_qnnpack():
    return
...
```

File: torch/testing/_internal/common_quantization.py [hunk 2]
```diff
@@ -785,19 +785,25 @@ def forward(self, x, hid):
         return x, hid
 
 class ConvModel(torch.nn.Module):
-    def __init__(self):
+    def __init__(self, transposed=False):
```
Comment (vkuzo) [inline torch/testing/_internal/common_quantization.py:788]: can we create a new class instead, to follow that `torch.nn.Conv` is a different class than `torch.nn.ConvTranspose`?

File: torch/testing/_internal/common_quantization.py [hunk 3]
```diff
@@ -785,19 +785,25 @@ def forward(self, x, hid):
...
+        else:
+            self.conv = torch.nn.Conv2d(3, 5, 3, bias=False).to(dtype=torch.float)
 
     def forward(self, x):
         x = self.conv(x)
         return x
 
 class AnnotatedConvModel(torch.nn.Module):
-    def __init__(self, qengine):
+    def __init__(self, qengine, transposed=False):
```
Comment (vkuzo) [inline torch/testing/_internal/common_quantization.py:800]: same here

File: test/quantization/test_quantize_jit.py [hunk 4]
```diff
@@ -3124,6 +3124,34 @@ def test_conv(self):
...
+            # eager mode
+            annotated_conv_model = AnnotatedConvModel(
+                torch.backends.quantized.engine, transposed=True).eval()
+            conv_model = ConvModel(transposed=True).eval()
```
Comment (vkuzo) [inline test/quantization/test_quantize_jit.py:3136]: creating a new `ConvTransposed` test class might be cleaner

**Review by vkuzo — APPROVED** [review]

**Pull request closed without merging**